### ISSUE TITLE
Get rid of hardcoded latency in FreqEstimator

### ISSUE THREAD

**Issue by gavv** [roc-project/roc#17]
Currently, latency can be configured at runtime for `Delayer`, but not for `FreqEstimator`. Thus, using non-default latency will break it.

**Comment by baranovmv** [comment]
To do so we need to build kind of a *latency_evaluator*. It is going to be a model of network behaviour, particularly model of sudden delays and losses.
I tried this and understood that it is easy to overfit the model to particular network environment. So before developing this we have to gather enough examples. So we need a good instrumentation for storing network behaviour.

**Comment by gavv** [comment]
Hmm.

1. Why do we need latency evaluator for *this* ticket? I'm talking about setting [`G_aim`](https://github.com/roc-project/roc/blob/master/src/modules/roc_audio/freq_estimator.cpp#L35) at runtime but only once (when FreqEstimator is created) instead of setting it in compile time. I'm not talking about adjusting `G_aim` dynamically (that is another story). Is it possible?

2. What instrumentation do we need to gather samples for latency evaluator? Related: #19. #20. Something else?

**Comment by baranovmv** [comment]
1. Ah, you want to add command line argument with goal value of latency? I see no obstacles for just setting [`G_aim`] before first call of [`roc::audio::FreqEstimator::update`]. You only have to accumulate right number of packets in the queue before run.

2. No, that is a good choice.

**Comment by gavv** [comment]
Good. So we need to make `G_aim` to be member of `FreqEstimator` configurable in constructor. `pipeline::Session` should pass the same latency value for `FreqEstimator` and `Delayer`.

But does `G_aim` depends somehow on value of [`FREQ_EST_DECIM_10_LEN`](https://github.com/roc-project/roc/blob/master/src/modules/roc_audio/freq_estimator_decim10_len.h#L17) or [this table](https://github.com/roc-project/roc/blob/master/src/modules/roc_audio/freq_estimator_decim10_coeff.h)?

**Comment by baranovmv** [comment]
No it doesn't.

**Comment by gavv** [comment]
Latency may be now configured using `--latency` option. However, too low values don't work because some minimal latency is required by `fec::Decoder`, `Interleaver` (if enabled on client) and audio output. I've added #44.

**Comment by gavv** [comment]
As for dynamic latency adjustment, it's in our [TODO](https://github.com/roc-project/roc/blob/master/TODO.md#advanced-features).
